### PR TITLE
Remove odo, networkx, and fix pytest fixtures

### PR DESCRIPTION
**Pull request by gwaybio**

### CODE REVIEW

**Comment by shntnu** [comment]
So awesome!! Is this WIP? @gwaygenomics

**Comment by gwaybio** [comment]
Yep, WIP - still need to check a couple things.

One immediate thing is my addition [here](https://github.com/cytomining/cytominer-database/pull/108/files#diff-9398b55cff89e452e5b4049aceaac50cR31).

It is required to pass tests b/c of lower case `image` in [`conftest.py`](https://github.com/cytomining/cytominer-database/blob/master/tests/conftest.py#L38). Not sure if this was intentional! Should `ingest` be case insensitive?

**Comment by gwaybio** [comment]
I removed appveyor and windows build checks in df425735fea61fa96015f8886c63741e3a49676c closes #109 

After merging, the PR closes #107

@shntnu - good to merge? (I will rebase before merging)

**Comment by shntnu** [comment]
> It is required to pass tests b/c of lower case `image` in [`conftest.py`](https://github.com/cytomining/cytominer-database/blob/master/tests/conftest.py#L38). Not sure if this was intentional! Should `ingest` be case insensitive?

I'd need to reorient myself to the code, but maybe #45 gives you clue?

**Comment by gwaybio** [comment]
Got it! Thanks for the pointer!

My read is that the key line is in [ingest.py](https://github.com/cytomining/cytominer-database/blob/master/cytominer_database/ingest.py#L151).

This capitalizes the `name` input in the `into()` function call. `into()` then creates the database named `target = "{}::{}".format(output, name)` where output is the connection string.

So when `test_ingest.py` calls [`seed()`](https://github.com/cytomining/cytominer-database/blob/master/tests/test_ingest.py#L19-L23) it will output a connection with a capitalized compartment.

## Decision

I guess then the decision is to change [`conftest.py`](https://github.com/cytomining/cytominer-database/blob/master/tests/conftest.py) or [`test_ingest.py`](https://github.com/cytomining/cytominer-database/pull/108/files#diff-0ea719227801cc8f0f167c332a35c22dR46). My vote would be to change conftest since it should mirror what we expect the output to be. 

I am not sure how this test was passing in the past! Perhaps the version of `odo` being used was not case sensitive? 🤔

File: cytominer_database/ingest.py [hunk 1]
```diff
@@ -69,7 +70,7 @@ def into(input, output, name, identifier, skip_table_prefix=False):
     """
 
     with backports.tempfile.TemporaryDirectory() as directory:
-        source = os.path.join(directory, os.path.basename(input))
+        source = os.path.join(os.path.abspath(directory), os.path.basename(input))
```
Comment (shntnu) [inline cytominer_database/ingest.py]: `os.path.abspath` is probably harmless, but could you clarify why you're doing it? @gwaygenomics

File: cytominer_database/ingest.py [hunk 2]
```diff
@@ -98,8 +99,12 @@ def into(input, output, name, identifier, skip_table_prefix=False):
             #     deprecated, use inspect.signature() or inspect.getfullargspec()
             warnings.simplefilter("ignore", category=DeprecationWarning)
```
Comment (shntnu) [inline cytominer_database/ingest.py:100]: Can we remove this filter now? @gwaygenomics

Comment (shntnu) [reply cytominer_database/ingest.py:100]: oh that's unrelated, ignore

File: cytominer_database/utils.py [hunk 3]
```diff
@@ -25,7 +25,13 @@ def find_directories(directory):
     :return: list of subdirectories of ``directory``
 
     """
-    return glob.glob(os.path.join(directory, '*/'))
+    directory = os.path.abspath(directory)
```
Comment (shntnu) [inline cytominer_database/utils.py]: Why was this needed? @gwaygenomics

File: tests/commands/test_command_ingest.py [hunk 4]
```diff
@@ -79,16 +78,15 @@ def test_run_defaults(cellpainting, runner):
         assert result.exit_code == 0
 
         for blob in cellpainting["ingest"]:
-            table_name = blob["table"]
+            table_name = blob["table"].capitalize()
```
Comment (shntnu) [inline tests/commands/test_command_ingest.py:81]: `.capitalize()` makes sense! 👍

File: tests/conftest.py [hunk 5]
```diff
@@ -19,7 +20,9 @@ def pytest_generate_tests(metafunc):
 def cellpainting():
     return {
         "config": "config.ini",
-        "data_dir": "tests/data_b",
+        "data_dir": os.path.join(
```
Comment (shntnu) [inline tests/conftest.py]: Why were these changes to specifying the path required? @gwaygenomics

File: tests/conftest.py [hunk 6]
```diff
@@ -98,13 +107,15 @@ def qc():
...
 @pytest.fixture
 def dataset(request):
-    if request.param == "htqc":
-        return htqc()
+    dataset_param = request.param
 
-    if request.param == "cellpainting":
-        return cellpainting()
+    if dataset_param == "htqc":
+        return request.getfixturevalue('htqc')
```
Comment (shntnu) [inline tests/conftest.py:109]: Could you please add a comment as to why we need to use `getfixturevalue`? @gwaygenomics

File: tests/test_ingest.py [hunk 7]
```diff
@@ -1,38 +1,42 @@
...
+            source=data_dir,
             target="sqlite:///{}".format(str(sqlite_file))
         )
 
-        for blob in dataset["ingest"]:
-            table_name = blob["table"]
-
-            csv_pathname = os.path.join(temp_dir, "{}.csv".format(table_name))
+        for blob in ingest:
+            table_name = blob["table"].capitalize()
```
Comment (shntnu) [inline tests/test_ingest.py:31]: `.capitalize()` here makes sense 👍

**Review by shntnu — COMMENTED** [review]
This looks great @gwaygenomics ! Could you address the comments below? We can pair next week if that's faster. Thanks for rescuing this!

**Comment by shntnu** [comment]
> I guess then the decision is to change [`conftest.py`](https://github.com/cytomining/cytominer-database/blob/master/tests/conftest.py) or [`test_ingest.py`](https://github.com/cytomining/cytominer-database/pull/108/files#diff-0ea719227801cc8f0f167c332a35c22dR46). My vote would be to change conftest since it should mirror what we expect the output to be.

I think changing `test_ingest.py` makes sense, because `conftest.py` reflects the real world problem where the file names may not be capitalized. Does that makes sense?

**Comment by shntnu** [comment]
> I am not sure how this test was passing in the past! Perhaps the version of `odo` being used was not case sensitive? 🤔

SQLite3 is not case sensitive

```
$ sqlite3
SQLite version 3.23.1 2018-04-10 17:39:29
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> .open ./inst/extdata/fixture_htqc.sqlite
sqlite> .tables
Cells      Cytoplasm  Nuclei     image
sqlite> select TableNumber from Image limit 10;
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
72e86570613007f2969abc4ce3771f3d
72e86570613007f2969abc4ce3771f3d
72e86570613007f2969abc4ce3771f3d
72e86570613007f2969abc4ce3771f3d
sqlite> select TableNumber from image limit 10;
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
5a822385194aceb347a07f27285dbfa3
72e86570613007f2969abc4ce3771f3d
72e86570613007f2969abc4ce3771f3d
72e86570613007f2969abc4ce3771f3d
```

File: cytominer_database/utils.py (hunk 3 above)
Comment (gwaybio) [reply cytominer_database/utils.py]: all changes to path arguments here were me troubleshooting the windows build 😅

File: tests/conftest.py (hunk 5 above)
Comment (gwaybio) [reply tests/conftest.py]: windows build troubleshooting. I can revert changes back!

File: tests/conftest.py (hunk 6 above)
Comment (gwaybio) [reply tests/conftest.py:109]: Good point - I will add!

File: cytominer_database/utils.py (hunk 3 above)
Comment (shntnu) [reply cytominer_database/utils.py]: 🥇 for trying!

**Comment by gwaybio** [comment]
> I think changing test_ingest.py makes sense, because conftest.py reflects the real world problem where the file names may not be capitalized. Does that makes sense?

I thought it over yesterday, and yeah, it does totally. We may want to test `conftest.py` in other functions and having a lowercase input mirrors real world problem

> Could you address the comments below? We can pair next week if that's faster. Thanks for rescuing this!

No problem! It is an essential component of morphological profiling research. Also, no need to pair - the comments are easy to address. More commits coming soon

**Comment by shntnu** [comment]
> No problem! It is an essential component of morphological profiling research. Also, no need to pair - the comments are easy to address. More commits coming soon

Super!

**Comment by gwaybio** [comment]
@shntnu - comments addressed and the build passed! Nothing more on my end, merge at will!

**Review by shntnu — APPROVED** [review]
💯
